pkg/cincinnati/cincinnati.go

```diff
@@ -51,14 +56,29 @@ func (err *Error) Error() string {
 	return fmt.Sprintf("%s: %s", err.Reason, err.Message)
 }
 
+// Run HTTP request in a goroutine and pass the response to f.
+func clientDo(ctx context.Context, req *http.Request, client http.Client, f func(*http.Response, error) error) error {
+	c := make(chan error, 1)
+	getUpdatesCtx, cancel := context.WithTimeout(ctx, getUpdatesTimeout)
+	defer cancel()
+	go func() { c <- f(client.Do(req.WithContext(getUpdatesCtx))) }()
+	select {
+	case <-getUpdatesCtx.Done():
+		<-c // Wait for f to return.
```

Comment:
But then you wait for `f` anyway?

Reply:
Correct. See https://github.com/openshift/cluster-version-operator/pull/410#discussion_r461198315 for my reasoning - right or wrong :)